controllers/signupController.js

```diff
@@ -19,53 +18,52 @@ signupController.checkEmailExistence = (req, res) => {
             res.send(err.stack);
         } else {
             if (result.rows[0] && !result.rows[0].is_deleted) {
-            res.send('emailExists');
+                res.send('emailExists');
             } else {
-            res.send('emailDoesntExist');
+                res.send('emailDoesntExist');
             }
         }
     });
 };
 
 signupController.register = (req, res) => {
-    let credentials = req.body;
+    const credentials = req.body;
     credentials.password = crypto.createHash('sha256').update(credentials.password).digest('hex');
 
-    // const client = new pg.Client(conString);
-    // client.connect();
     db.query(signupModel.upsertIntoUsers(credentials.email, credentials.phone, credentials.password),
-    (err,result) => {
+    (err, result) => {
         if (err) {
             console.log(err);
         } else {
-            db.query(signupModel.getId(credentials.email), (err,res) => {
+            db.query(signupModel.getId(credentials.email), (err, resInner) => {
                 if (err) {
                     console.log(err);
                 } else {
-                    const confirmStr = crypto.createHash('sha1').update(''+res.rows[0].id).digest('hex')+res.rows[0].id;
+                    const confirmStr = crypto.createHash('sha1').update('' + resInner.rows[0].id).digest('hex') + resInner.rows[0].id;
                     const fullUrl = `${req.protocol}://${req.get('host')}${req.originalUrl}/${confirmStr}`;
```

Comment:
if i know an id I can hack it.
Please generate `nonsense` and save into db then check `nonsense` and code user uses for verification.